test/test_json_query_parser.cpp

```diff
@@ -0,0 +1,468 @@
+
+#include "test.hpp"
+#include <realm/parser/driver.hpp>
+#include <realm/table_view.hpp>
+
+using namespace realm;
+using json = nlohmann::json;
+
+// basic consts and props
+json int_const = {{"kind", "constant"}, {"value", 3}, {"type", "int"}};
+json string_const = {{"kind", "constant"}, {"value", "Bob"}, {"type", "string"}};
+json float_const = {{"kind", "constant"}, {"value", 2.22}, {"type", "float"}};
+json long_const = {{"kind", "constant"}, {"value", LONG_MAX}, {"type", "long"}};
+json double_const = {{"kind", "constant"}, {"value", 2.2222222}, {"type", "double"}};
+json bool_const_true = {{"kind", "constant"}, {"value", true}, {"type", "bool"}};
+json bool_const_false = {{"kind", "constant"}, {"value", false}, {"type", "bool"}};
+json int_prop = {{"kind", "property"}, {"name", "age"}, {"type", "int"}};
+json string_prop = {{"kind", "property"}, {"name", "name"}, {"type", "string"}};
+json float_prop = {{"kind", "property"}, {"name", "fee"}, {"type", "float"}};
+json long_prop = {{"kind", "property"}, {"name", "salary"}, {"type", "long"}};
+json double_prop = {{"kind", "property"}, {"name", "longitude"}, {"type", "double"}};
+json bool_prop = {{"kind", "property"}, {"name", "isInteresting"}, {"type", "bool"}};
```

Comment:
To make sure links are supported, I'd like to see some tests for properties across links such as `buddy.age > 2`

Reply:
links are not supported yet either!